dot/network/utils_test.go

```diff
@@ -5,22 +5,35 @@ package network
 
 import (
 	"bytes"
+	"net"
 	"testing"
 
 	"github.com/ChainSafe/gossamer/lib/utils"
 	"github.com/stretchr/testify/require"
 )
 
+func availablePort(t *testing.T) uint16 {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	require.NoError(t, err)
+
+	l, err := net.ListenTCP("tcp", addr)
+	require.NoError(t, err)
+	defer l.Close()
+
+	port := l.Addr().(*net.TCPAddr).Port
+	return uint16(port)
```

Comment:
got it! I was thinking about: first check if the default port is available otherwise get a free port from OS using the `*:0`

Reply:
You need to bind to an available port at the same time as you request it; otherwise if you release it back to the OS it might get binded by something else. Although now that I think about it, I'm not 100% sure how the OS handles port `0` for more than 2^16 parallel tests for example. Does it just block or does it panic 🤔 Anyway I doubt we would reach that many tests 😄